web/src/features/modals/modalAtoms.ts

```diff
@@ -1,5 +1,27 @@
 import { atom } from 'jotai';
 
-export const isFAQModalOpenAtom = atom(false);
-export const isInfoModalOpenAtom = atom(false);
-export const isSettingsModalOpenAtom = atom(false);
+type ModalType = 'settings' | 'layers' | null;
+
+const activeModalAtom = atom<ModalType>(null);
+
+export const isSettingsModalOpenAtom = atom(
+  (get) => get(activeModalAtom) === 'settings',
+  (get, set, value?: boolean) => {
+    if (value === false) {
+      set(activeModalAtom, null);
+    } else {
+      set(activeModalAtom, get(activeModalAtom) === 'settings' ? null : 'settings');
+    }
+  }
+);
+
+export const isLayersModalOpenAtom = atom(
+  (get) => get(activeModalAtom) === 'layers',
+  (get, set, value?: boolean) => {
+    if (value === false) {
+      set(activeModalAtom, null);
+    } else {
+      set(activeModalAtom, get(activeModalAtom) === 'layers' ? null : 'layers');
+    }
+  }
+);
```

Comment:
```suggestion
// Factory function to create modal atoms
const createModalAtom = (modalType: NonNullable<ModalType>) =>
  atom(
    (get) => get(activeModalAtom) === modalType,
    (get, set, value?: boolean) => {
      if (value === false) {
        // Close only if this modal is currently open
        if (get(activeModalAtom) === modalType) {
          set(activeModalAtom, null);
        }
      } else {
        // Toggle behavior or explicitly open
        set(activeModalAtom, get(activeModalAtom) === modalType ? null : modalType);
      }
    }
  );

export const isSettingsModalOpenAtom = createModalAtom('settings');
export const isLayersModalOpenAtom = createModalAtom('layers');
```
This creates a factory and removes any code duplication. It also somewhat removes the need for the constants for layers and settings.

Reply:
I kinda prefer the duplicated version here